static/js/user_profile.js

```diff
@@ -28,6 +28,9 @@ import * as user_groups from "./user_groups";
 import * as user_pill from "./user_pill";
 import * as util from "./util";
 
+let list_of_unsubscribed_streams = [];
+let is_unsubscribed = false;
```

Comment:
This being a global variable doesn't make sense to me. 

Structurally, I think the bug is likely to be just that we need to trigger a rerender of any open "Streams" panel of a user profile when an event updates a user's subscriptions via `stream_events.js` -- you shouldn't need to do anything in this file other than having such a hook available, `stream_data.get_subscribed_streams_for_user` should return the right thing if using current data.